components/dashboard/src/settings/SelectIDEModal.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useState, useContext } from "react";
+import { Link } from "react-router-dom";
+import SelectIDE from "./SelectIDE";
+import Modal from "../components/Modal";
+
+import { getGitpodService } from "../service/service";
+import { UserContext } from "../user-context";
+
+export interface SelectEditorModal {
+    showModal?: boolean;
+}
+
+export default function (props: SelectEditorModal) {
+    const [visible, setVisible] = useState(props.showModal ?? false);
+    const { user, setUser } = useContext(UserContext);
+
+    const handleContinue = async () => {
```

Comment:
We can use `SelectIDE` component here, and delete these code, since they already exist in `SelectIDE`

Reply:
Every user selection will change user setting in DB, we can just wait until they click `continue`. It will make code easier

Reply:
Deal with re-use login in `SelectIDE` and will only be called if user not select any of them👍 cc @andreafalzetti